src/orquestra/sdk/_base/_spaces/_resolvers.py

```diff
@@ -0,0 +1,74 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+
+"""
+Tools for handling combinations of workspace and project.
+"""
+
+import typing as t
+
+from orquestra.sdk.schema.workflow_run import ProjectId, WorkspaceId
+
+from .._spaces._structs import Project, ProjectRef, Workspace
+
+
+def get_space_ids(
+    workspace: t.Optional[t.Union[Workspace, WorkspaceId]],
+    project: t.Optional[t.Union[Project, ProjectRef, ProjectId]],
```

Comment:
**nitpick**: you're probably aware of it, but `Workspace`, `Project`, and `ProjectRef` cases won't be needed here after `sdk.list_workflow_runs()` is simplified to accept just `Optional[WorkspaceId]` and `Optional[ProjectId]`

Reply:
Get_space_ids is removed in [99587a9](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/153/commits/99587a9a288ab991de54a6414336d8ad42acd7d7) as we no longer need to worry about clashes.